src/common/src/types/mod.rs

```diff
@@ -640,6 +640,58 @@ impl ScalarRefImpl<'_> {
     }
 }
 
+/// `for_all_cmp_types` helps in matching and casting types when building comparison expressions
+///  such as <= or IS DISTINCT FROM.
+#[macro_export]
+macro_rules! for_all_cmp_variants {
```

Comment:
Can we keep this macro in `crate expr` rather than in `crate common`?

Reply:
Sure. Would `/expr/src/expr/template.rs` be a suitable place for it?